### PR TITLE
units rework

### PR DESCRIPTION
**Pull request by mcneish1**
This PR switches GridDyn over to the new units implementation.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "ThirdParty/units"]
+	path = ThirdParty/units
+	url = https://github.com/mcneish1/units
```
Comment (phlptp) [inline .gitmodules:3]: is this going to work since it is a private repo yet?

Comment (mcneish1) [reply .gitmodules:3]: Yep, when you update your submodules, it should prompt you for a github.com password

File: src/fmi/fmi_models/fmiCoSimSubModel.cpp [hunk 2]
```diff
@@ -304,7 +306,7 @@ void fmiCoSimSubModel::set (const std::string &param, double val, gridUnits::uni
         }
         else
         {
-            gridSubModel::set (param, val);
+            gridSubModel::set (param, val); // TODO isn't this where the unitType is supposed to go?
```
Comment (phlptp) [inline src/fmi/fmi_models/fmiCoSimSubModel.cpp:309]: probably

File: .gitmodules (hunk 1 above)
Comment (phlptp) [reply .gitmodules:3]: how will that work with Travis?

Comment (phlptp) [reply .gitmodules:3]: might be best to just stick the code in a thirdParty folder and build it with the utilities library until the library gets a little more settled.

**Comment by phlptp** [comment]
Should we make a new branch in the GridDyn Repo to merge this until we get it all working, that might be simpler than doing it in the cmake_updates branch

File: .gitmodules (hunk 1 above)
Comment (mcneish1) [reply .gitmodules:3]: There's no issue once units is allowed to be public. Travis is probably cloning recursively already, and if not that's an easy thing to fix.

**Comment by mcneish1** [comment]
> Should we make a new branch in the GridDyn Repo to merge this until we get it all working, that might be simpler than doing it in the cmake_updates branch

This _is_ a new branch. It's targeting cmake_updates because that's the most up-to-date one

**Comment by phlptp** [comment]
I don't want to merge into cmake_update until the tests pass on travis, and I don't think that is possible while the repo is private

**Comment by mcneish1** [comment]
What's the hurry merging this into anything? I don't understand the purpose of merging this into a different temporary branch pre-cmake_update.

**Comment by mcneish1** [comment]
Merged into helics_updates

**Pull request closed without merging**